cmake/Modules/Findelsi.cmake

```diff
@@ -0,0 +1,34 @@
+if(NOT TARGET elsi::elsi)
+  find_package(PkgConfig QUIET)
+  pkg_check_modules(ELSI QUIET elsi)
+  if(ELSI_FOUND)
+    message(STATUS "Found 'elsi' via pkg-config")
+
+    add_library(elsi::elsi INTERFACE IMPORTED)
+    target_link_libraries(
+      elsi::elsi
+      INTERFACE
+      "${ELSI_LINK_LIBRARIES}"
+    )
+    target_include_directories(
+      elsi::elsi
+      INTERFACE
+      "${ELSI_INCLUDE_DIRS}"
+    )
+
+    if("${ELSI_LINK_LIBRARIES}" MATCHES "pexsi")
+      add_library(elsi::pexsi INTERFACE IMPORTED)
```

Comment:
I'd suggest to define an explicit dependency as well:
```
target_link_library(elsi::pexsi INTERFACE elsi::elsi)
```
So linking against `elsi::pexsi` only would also automatically add the right linking directives.